command_server/controllers/main.py

```diff
@@ -144,61 +157,57 @@ def set_invaders(self):
+
+    def bullet_hero(self):
+        move_bullet = True
+        while move_bullet:
+            self.hero.move_bullet()
+            for _invader in self.Invaders:
+                if self.hero.check_fire(_invader, self.game_field['height']):
                     move_bullet = False
-                    self.Invaders[unit_number].bullet = Bullet(self.Invaders[unit_number], self.Invaders[unit_number].bullet.image_filename)
-        if action == 'bullet_hero':
-            move_bullet = True
-            while move_bullet:
-                self.hero.move_bullet()
-                # print 'bullet - ', game_object.hero.bullet.__dict__
-                for _invader in self.Invaders:
-                    if self.hero.check_fire(_invader, self.game_field['height']):
-                        move_bullet = False
-                        self.hero.bullet = Bullet(self.hero, self.hero.bullet.image_filename)
-        if action == 'move_right':
-            self.hero.move_to(self.hero.x + self.hero.speed, self.hero.y)
-        if action == 'move_left':
-            self.hero.move_to(self.hero.x - self.hero.speed, self.hero.y)
-
-    def run(self):
-        game_object = GameController(50, 50, 2)
-
-        '''this code for moving invaders. Work as a job.
-            We set moving_speed for positive - if reach the left coordinate of our game field
-            or negative  - if we reach the right coordinate of our game field '''
-
-        while not game_object.hero.is_dead and len(game_object.Invaders):
-            game_object.Invaders[0].set_speed(game_object.game_field['width'])
-            game_object.Invaders[-1].set_speed(game_object.game_field['width'])
-            check_if_move_y = game_object.Invaders[0].check_if_move_y()
-            random_number = randint(0, len(game_object.Invaders) - 1)
-            game_object.get_action('bullet_invader', random_number)
-            for invader in game_object.Invaders:
-                new_x = invader.x + invader.moving_speed
-                new_y = check_if_move_y and invader.y + invader.speed or invader.y
-                invader.move_to(new_x, new_y)
-
-            # for invader in game_object.Invaders:
-            #     print invader.__dict__
-
-            time.sleep(3)
-
-if __name__ == "__main__":
-
-    '''This example of calling actions such as 'bullet', 'move_left' for our hero
-       Also we can call action 'move_right' '''
-
-    # game_object = GameController(50, 50, 2)
-    # game_object.get_action('bullet')
-    # game_object.get_action('move_left')
-    # game_object.get_action('move_left')
-
-    # for invader in game_object.Invaders:
-    #     print 'invader - ', invader.__dict__
-    # print 'hero - ', game_object.hero.__dict__
+                    self.hero.bullet = Bullet(self.hero, self.hero.bullet.image_filename)
+
+    def move_right(self):
+        self.hero.move_to(self.hero.x + self.hero.speed, self.hero.y)
```

Comment:
speed is a value based on time. You cannot just add speed to coordinates. I'd recommend to use `sleep` based on the speed and compare time between previous location and the new. If it's passed two seconds (of `sleep`) for example, then you take the speed into account - how many steps can be made during that second. For the speed of 5 steps/sec you'd have to do `hero.x` + 5 * 2 (seconds). 

Let's assume each step or move we made is some single action after which we do a `sleep(0.5)` for continuous move or just stop if reached target location. With continuous loop it means that each 0.5s we will be running the method to update unit position based on its previous location, time passed and current speed